javalib/src/main/scala/java/net/Socket.scala

```diff
@@ -145,8 +144,8 @@ class Socket protected (
 
     checkClosedAndCreate()
 
+    impl.bind(addr.getAddress, addr.getPort)
```

Comment:
Sorry, I do not understand the benefit of the slight change.

Does it bring any benefit that warrants touching a file which
otherwise need not change to support Datagrams?

Reply:
This change is not related to the datagram. I just modified this so when `bind` throws, the internal state is not be modified. Previously, the `localAddr` is mutated before the operation. In the case when the exception is caught, we end up with a undound socket with improper `localAddr`.
I'll revert and include this in a separate PR.